lib/improver/orographic_enhancement.py

```diff
@@ -0,0 +1,600 @@
+
+    def process(self, temperature, humidity, pressure, uwind, vwind,
+                topography, svp_method='IMPROVER'):
+        """
+        Calculate precipitation enhancement over orography on standard (2 km)
+        and high resolution (1 km UKPP domain) grids
+
+        Args:
+            temperature (iris.cube.Cube):
+                Temperature at top of boundary layer
+            humidity (iris.cube.Cube):
+                Relative humidity at top of boundary layer
+            pressure (iris.cube.Cube):
+                Pressure at top of boundary layer
+            uwind (iris.cube.Cube):
+                Positive eastward wind vector component at top of boundary
+                layer
+            vwind (iris.cube.Cube):
+                Positive northward wind vector component at top of boundary
+                layer
+            topography (iris.cube.Cube):
+                Height of topography above sea level on 1 km UKPP domain grid
+
+        Kwargs:
+            svp_method (str):
+                Which method to use to calculate the saturation vapour
+                pressure (IMPROVER or STEPS).  Defaults to using the IMPROVER
+                WetBulbTemperature plugin.
+
+        Returns:
+            (tuple): tuple containing:
+                **orogenh** (iris.cube.Cube):
+                    Precipitation enhancement due to orography in mm/h on the
+                    1 km Transverse Mercator UKPP grid domain
+                **orogenh_standard_grid** (iris.cube.Cube):
+                    Precipitation enhancement due to orography in mm/h on the
+                    2 km standard grid
+
+        References:
+            Alpert, P. and Shafir, H., 1989: Meso-Gamma-Scale Distribution of
+                Orographic Precipitation: Numerical Study and Comparison with
+                Precipitation Derived from Radar Measurements.  Journal of
+                Applied Meteorology, 28, 1105-1117.
+            Roe, G., 2005: Orographic Precipitation.  Annual Review of Earth
+                and Planetary Sciences, 33, 645-671.
+        """
+        # check input variable cube coordinates match
+        unmatched_coords = compare_coords(
+            [temperature, pressure, humidity, uwind, vwind])
+
+        if any(item.keys() for item in unmatched_coords):
+            msg = 'Input cube coordinates {} are unmatched'
+            raise ValueError(msg.format(unmatched_coords))
+
+        # check all cubes are 2D spatial fields
+        msg = 'Require 2D fields as input; found {} dimensions'
+        if temperature.ndim > 2:
+            raise ValueError(msg.format(temperature.ndim))
+        check_for_x_and_y_axes(temperature)
+
+        if topography.ndim > 2:
+            raise ValueError(msg.format(topography.ndim))
+        check_for_x_and_y_axes(topography)
+
+        # check pressure cube units (iris doesn't recognise 'mb')
+        if pressure.units == Unit('mb'):
+            pressure.units = Unit('hPa')
```

Comment:
This is only relevant for the test data suppled from the old post processing system. A check like this belongs in the CLI, if anywhere. Please remove it.

Reply:
I feel like this is a "gotcha" that should be captured somewhere.  Is it worth adding explicitly to #599?